### DISCUSSION TITLE
DAPimpleDyMFoam to simulate dynamicMesh motion

### DISCUSSION THREAD

**Discussion by wndzjl** [mdolab/dafoam#702] [Tutorials]
I wanted to do unsteady aerodynamic optimization of sink and float airfoil, but I couldn't find a tutorial on this solver. I would like to know about the relevant Settings and introduction of this solver, and it is best to have a case.

**Comment by friedenhe** [comment]
The dynamicMesh capability is not available for design optimization yet, it can do flow analysis only. We plan to add this capability sometime next year. Stay tuned.

**Reply by wndzjl** [reply]
I read in the openfoam tutorial that can use pimplefoam to simulate pitching airfoils. Do you mean DAPimpleDyMFoam does not support dynamicMesh or pimple and pisofoam do not support it either? I wonder if you could share the NACA0012 case code in the article "A Hybrid Time-Spectral Approach for Aerodynamic Optimization with Unsteady Flow"? Thank you very much！https://wiki.openfoam.com/Dynamic_stall_NACA0012_Michael_Alletto

**Reply by wndzjl** [reply]
What types of unsteady optimizations can DAfoam do now? Dynamic mesh are not supported, but are sliding mesh supported? I particularly want to optimize the pitch variation airfoil, is it possible to achieve this in DAfoam by other methods? Thank you very much!

**Comment by friedenhe** [comment]
Yes, you can use pimpleFoam to "simulate" pitching airfoil, but you can't use it to conduct adjoint-based "optimization" because the adjoint solver has not been implemented for pimpleDymFoam.. Sliding mesh is not supported either.. No, if you want to consider pitching airfoil cases,  you need the dynamic mesh capability..

**Reply by 101709** [reply]
So what types of unsteady optimizations can DAfoam do now? Thank you!

**Reply by wndzjl** [reply]
Thank you very much for Professor He’s patient response. I have one last small wish: could Professor He share the last two cases from the paper _“A Hybrid Time-Spectral Approach for Aerodynamic Optimization with Unsteady Flow”_? I have encountered many questions during my studies and would hope to learn more about the results you have achieved. Wish you everlasting achievements in academia.

**Reply by wndzjl** [reply]
Dear Professor He,
Can the latest dafoam realize the shape optimization function of the pitch airfoil?
If so, is there any reference case? What type of dynamic mesh is DApimpleDymFoam used?

**Reply by friedenhe** [reply]
@wndzjl We have recently added a dynamic mesh flow and adjoint solver (DAPimpleDymFoam), but we are still testing its capability for pitching airfoil optimization. No reference case is available at this moment, but we should be able to have one sometime in the summer.

**Comment by friedenhe** [comment]
DAFoam can do unsteady optimization with static boundaries now. The unsteady optimization capability for dynamic boundaries will be added in 2025. I don't have the data or case setup for "A Hybrid Time-Spectral Approach for Aerodynamic Optimization with Unsteady Flow”" because it was based on a very old version of DAFoam...

**Reply by wndzjl** [reply]
OK, thank you so much!
